src/persistence/data_access.py

```diff
@@ -1,5 +1,121 @@
-# TODO: Creat DataAccess class, inherit from IDataAccess. Class must contain the following private variables:
-#  - __app_data_path:   str - Consider using app data directory.
+from idata_access import IDataAccess
+import os, json
 
-# TODO: Create constructor for DataAccess class, that sets all variables above.
-# TODO: Implement the abstract methods: save_config(runnable: FileInfo, data: dict), load_config(runnable: FileInfo), save_main_runnables(runnables: dict), load_main_runnables(), clear_history()
+"""
+This module provides a class DataAccess that handles data persistence for the application.
+It inherits from IDataAccess and implements the abstract methods for saving and loading configurations.
+"""
+
+from idata_access import IDataAccess
+import os, json
+
+class DataAccess(IDataAccess):
+    """
+    A class that handles data persistence for the application.
+
+    Attributes:
+    __app_data_path (str): The path to the application data directory.
+
+    Methods:
+    __init__(data_path: str): Initializes the DataAccess object with the application data path.
+    save_config(runnable: str, data: dict): Saves the configuration for a runnable.
+    load_config(runnable: str) -> dict: Loads the configuration for a runnable.
+    save_main_runnables(data: dict): Saves the main runnables.
+    load_main_runnables() -> dict: Loads the main runnables.
+    clear_history(): Clears the configuration history.
+    """
+
+    def __init__(self, data_path: str):
+        """
+        Initializes the DataAccess object with the application data path.
+
+        Args:
+        data_path (str): The path to the application data directory.
+        """
+        self.__app_data_path = os.path.join(data_path, "SZOFTECH")
+
+    def save_config(self, runnable: str, data: dict):
+        """
+        Saves the configuration for a runnable.
+
+        Args:
+        runnable (str): The name of the runnable.
+        data (dict): The configuration data.
+        """
+        if not os.path.exists(self.__app_data_path):
+            os.makedirs(self.__app_data_path, exist_ok=True)
+        file_name = f"{runnable}.json" 
```

Comment:
if runnable's string contains '/' or '\' please replace it with some other character '_'